walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,95 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
+
+  /** 
+   * Return all possible time ranges to hold a meeting of specific duration.
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Find which times work for mandatory attendees
+    ArrayList<TimeRange> busyRangesForMandatoryAttendees = findBusyRanges(request.getAttendees(), events);
+    Collections.sort(busyRangesForMandatoryAttendees, TimeRange.ORDER_BY_START);
+    Collection<TimeRange> freeRangesForMandatoryAttendees = findFreeRanges(request, busyRangesForMandatoryAttendees);
```

Comment:
Why does a similar method above return an ArrayList while this returns a Collection?

Reply:
If you made this Collection because the return type of the method is Collection, then you can return a subclass instead. So you can return a List or ArrayList if the return type is Collection.